pulsar-transformations/src/main/java/com/datastax/oss/pulsar/functions/transforms/jstl/JstlTransformContextAdapter.java

```diff
@@ -115,4 +113,28 @@ public Object getValue() {
   public Map<String, Object> getHeader() {
     return lazyHeader;
   }
+
+  /** Enables {@link LazyMap} lookup on {@link GenericRecord}. */
+  static class GenericRecordTransformer implements Transformer<String, Object> {
+
+    GenericRecord genericRecord;
+
+    public GenericRecordTransformer(GenericRecord genericRecord) {
+      this.genericRecord = genericRecord;
+    }
+
+    @Override
+    public Object transform(String key) {
+      Object value = null;
+      if (genericRecord.hasField(key)) {
+        value = genericRecord.get(key);
+        if (value instanceof GenericRecord) {
```

Comment:
for the future:
what about "arrays" ?
this is not top priority, but we should at least create a GH issue

Reply:
This code in particular is to support nested access. Arrays already supported by the syntax. I will few unit tests for it in the next patch.